internal/service/encryptionatrest/resource_encryption_at_rest_test.go

```diff
@@ -5,101 +5,22 @@ import (
 	"errors"
 	"fmt"
 	"os"
+	"strconv"
 	"testing"
 
+	"go.mongodb.org/atlas-sdk/v20240805001/admin"
+	"go.mongodb.org/atlas-sdk/v20240805001/mockadmin"
+
 	"github.com/hashicorp/terraform-plugin-framework/types"
 	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
 	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/retrystrategy"
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/service/encryptionatrest"
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/acc"
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/mock"
-	"go.mongodb.org/atlas-sdk/v20240805001/admin"
-	"go.mongodb.org/atlas-sdk/v20240805001/mockadmin"
-)
-
-const (
-	initialConfigEncryptionRestRoleAWS = `
-provider "aws" {
-	region     = lower(replace("%[1]s", "_", "-"))
-	access_key = "%[2]s"
-	secret_key = "%[3]s"
-}
-
-%[7]s
-
-resource "mongodbatlas_cloud_provider_access" "test" {
```

Comment:
`mongodbatlas_cloud_provider_access` resource doesn't exist anymore so this test hasn't been working for a while. Updated the config and moved to `testAccMongoDBAtlasEncryptionAtRestConfigAwsKmsWithRole` which creates all required resources.